lib/src/api/api.dart

```diff
@@ -16,7 +16,10 @@ class ArweaveApi {
       _client.get(_getEndpointUri(endpoint));
 
   Future<http.Response> post(String endpoint, {dynamic body}) =>
-      _client.post(_getEndpointUri(endpoint), body: body);
+      _client.post(_getEndpointUri(endpoint), body: body).catchError((onError) {
+        print('Caught error in post');
+        return http.Response('', 404);
```

Comment:
Why 404? I'm not sure, but It could be 5xx as well, right?
Does `onError` provide the http status and possibly the error message?

Reply:
This update isn't necessary given the way you've wrapped post below.

Reply:
Yes, we get the `statusCode` on `lastResponseStatus`, but would be good to have meaningful error messages here as well, it can help debugging.